### PR TITLE
Extract utility functions

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

Extracted common utility functions related to JSON parsing
Will wait for #149 to be merged, and then put this up for review.

#### Motivation and Context

Remove code duplication

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/internal-server/app/Controllers/Http/UserController.js [hunk 1]
```diff
@@ -304,25 +299,17 @@ class UserController {
         .rows.length > 0
     );
 
-    if (
-      workspace.services &&
-      typeof workspace.services === 'string' &&
-      workspace.services.length > 0
```
Comment (SpecialAro) [inline src/internal-server/app/Controllers/Http/UserController.js:310]: This is no longer necessary?

File: src/internal-server/app/Controllers/Http/UserController.js [hunk 2]
```diff
@@ -304,25 +299,17 @@ class UserController {
...
     const services =
       workspace.services && typeof workspace.services === 'object'
         ? workspace.services.map(
             oldServiceId => serviceIdTranslation[oldServiceId],
           )
         : [];
-    if (
-      workspace.data &&
-      typeof workspace.data === 'string' &&
-      workspace.data.length > 0
```
Comment (SpecialAro) [inline src/internal-server/app/Controllers/Http/UserController.js:323]: Same here (?)

File: src/internal-server/app/Controllers/Http/UserController.js [hunk 3]
```diff
@@ -347,12 +334,8 @@ class UserController {
     // store the old serviceId as the key for future lookup
     serviceIdTranslation[service.serviceId] = newServiceId;
 
-    if (
-      service.settings &&
-      typeof service.settings === 'string' &&
-      service.settings.length > 0
```
Comment (SpecialAro) [inline src/internal-server/app/Controllers/Http/UserController.js:353]: And here (?)

**Review by SpecialAro — APPROVED** [review]
Overall LGTM!

File: src/internal-server/app/Controllers/Http/UserController.js (hunk 1 above)
Comment (vraravam) [reply src/internal-server/app/Controllers/Http/UserController.js:310]: thanks - must have slipped my sight. will fix in new commit

Comment (vraravam) [reply src/internal-server/app/Controllers/Http/UserController.js:310]: that's already handled in the `convertJSON` method

File: src/internal-server/app/Controllers/Http/UserController.js (hunk 3 above)
Comment (vraravam) [reply src/internal-server/app/Controllers/Http/UserController.js:353]: that's already handled in the `convertJSON` method

File: src/internal-server/app/Controllers/Http/UserController.js (hunk 2 above)
Comment (vraravam) [reply src/internal-server/app/Controllers/Http/UserController.js:323]: that's already handled in the `convertJSON` method

**Pull request merged**